src/redux/books/books.js

```diff
@@ -1,27 +1,43 @@
-const ADD_BOOK = 'book-store/books/ADD_BOOK';
-const REMOVE_BOOK = 'book-store/books/REMOVE_BOOK';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import BookApi from '../../Api_service/BooksService';
 
-const books = [];
+const ADD_BOOK = 'books/ADD_BOOK';
+const REMOVE_BOOK = 'books/REMOVE_BOOK';
+const LOAD_BOOKS = 'books/LOAD_BOOKS';
 
-const BookReducer = (state = books, action) => {
-  switch (action.type) {
-    case ADD_BOOK:
-      return state.push(action.book);
-    case REMOVE_BOOK:
-      return state.filter((book) => book.id !== action.id);
-    default:
-      return state;
+export const loadBooks = createAsyncThunk(LOAD_BOOKS, async () => {
+  try {
+    const res = await BookApi.getBookApi();
+    return res.data;
+  } catch (error) {
+    throw new Error(error);
   }
-};
+});
 
-export const addBook = (book) => ({
-  type: ADD_BOOK,
-  book,
+export const addBook = createAsyncThunk(ADD_BOOK, async (payload, thunkAPI) => {
```

Comment:
- [ ] I kindly suggest that `addBook` and `removeBook` should still work after implementing your API. Kindly see the video below for more details:

https://www.loom.com/share/556223a85ead40bb85b1c874c34c7b52